StartioAdmobMediation/StartioAdmobCustomEvent.m

```diff
@@ -0,0 +1,146 @@
+                    [self setupStartioSDKWithMediationParameters:mediationParameters testAdsEnabled:NO];
+                });
+                completionHandler(nil);
+            }
+        }
+        else {
+            completionHandler([NSError errorWithDomain:GADErrorDomain code:GADErrorMediationAdapterError userInfo:@{NSLocalizedDescriptionKey : @"No Start.io AppID provided yet"}]);
+        }
+    }
+}
+
++ (void)setupStartioSDKWithMediationParameters:(StartioAdmobParameters *)mediationParameters testAdsEnabled:(BOOL)testAdsEnabled {
+    STAStartAppSDK *sdk = [STAStartAppSDK sharedInstance];
+    
+    sdk.appID = mediationParameters.appId;
+    sdk.returnAdEnabled = NO;
+    sdk.consentDialogEnabled = NO;
+    [sdk addWrapperWithName:@"AdMob" version:StartioAdmobAdapterVersion];
+    sdk.testAdsEnabled = YES;
+#ifdef DEBUG
+    sdk.testAdsEnabled = YES;
+#else
+    sdk.testAdsEnabled = testAdsEnabled;
+#endif
+}
+
+- (void)loadBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationBannerLoadCompletionHandler)completionHandler {
+    self.bannerAdapter = [[StartioAdmobBannerAdapter alloc] init];
+    [self loadAdForAdapter:self.bannerAdapter adConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
+- (void)loadInterstitialForAdConfiguration:(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)completionHandler {
+    self.interstitialAdapter = [[StartioAdmobInterstitialAdapter alloc] init];
+    [self loadAdForAdapter:self.interstitialAdapter adConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
+- (void)loadNativeAdForAdConfiguration:(nonnull GADMediationNativeAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler {
+    self.nativeAdapter = [[StartioAdmobNativeAdapter alloc] init];
+    [self loadAdForAdapter:self.nativeAdapter adConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
+- (void)loadRewardedAdForAdConfiguration:(nonnull GADMediationRewardedAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationRewardedLoadCompletionHandler)completionHandler {
+    self.rewardedAdapter = [[StartioAdmobRewardedAdapter alloc] init];
+    [self loadAdForAdapter:self.rewardedAdapter adConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
+- (void)loadAdForAdapter:(nonnull id<StartioAdmobAdAdapter>)adapter adConfiguration:(nonnull GADMediationAdConfiguration *)adConfiguration completionHandler:(nonnull id)completionHandler {
+    StartioAdmobParameters *parameters = [[StartioAdmobParameters alloc] initWithParametersJSON:adConfiguration.credentials.settings[kSettingsParameterKey]];
+    [StartioAdmobCustomEvent setupStartioSDKWithMediationParameters:parameters testAdsEnabled:adConfiguration.isTestRequest];
```

Comment:
why each time new request is performed the setup is performed to? I would expect this is performed on explicit call to setUpWithConfiguration:completionHandler: only